roles/tests-integration/tests/common/mod.rs

```diff
@@ -147,8 +166,130 @@ impl TemplateProvider {
+    pub pool: PoolSv2,
+}
+
+impl TestPoolSv2 {
+    pub fn new(
+        listening_address: Option<SocketAddr>,
+        coinbase_outputs: Option<Vec<pool_sv2::mining_pool::CoinbaseOutput>>,
+        template_provider_address: Option<SocketAddr>,
+    ) -> Self {
+        use pool_sv2::mining_pool::{CoinbaseOutput, Configuration};
+        let pool_port = if let Some(listen_addr) = listening_address {
+            listen_addr.port()
+        } else {
+            get_available_port()
+        };
+        let listening_address = listening_address
+            .unwrap_or(SocketAddr::from_str(&format!("127.0.0.1:{}", pool_port)).unwrap());
+        let is_pool_port_open = is_port_open(listening_address);
+        assert!(!is_pool_port_open);
+        let authority_public_key = Secp256k1PublicKey::try_from(
+            "9auqWEzQDVyd2oe1JVGFLMLHZtCo2FFqZwtKA5gd9xbuEu7PH72".to_string(),
+        )
+        .expect("failed");
+        let authority_secret_key = Secp256k1SecretKey::try_from(
+            "mkDLTBBRxdBv998612qipDYoTK3YUrqLe8uWw7gu3iXbSrn2n".to_string(),
+        )
+        .expect("failed");
+        let cert_validity_sec = 3600;
+        let coinbase_outputs = if let Some(cb_outs) = coinbase_outputs {
+            cb_outs
+        } else {
+            vec![CoinbaseOutput::new(
+                "P2WPKH".to_string(),
+                "036adc3bdf21e6f9a0f0fb0066bf517e5b7909ed1563d6958a10993849a7554075".to_string(),
+            )]
+        };
+        let pool_signature = "Stratum v2 SRI Pool".to_string();
+        let tp_address = if let Some(tp_add) = template_provider_address {
+            tp_add.to_string()
+        } else {
+            "127.0.0.1:8442".to_string()
+        };
+        let connection_config = pool_sv2::mining_pool::ConnectionConfig::new(
+            listening_address.to_string(),
+            cert_validity_sec,
+            pool_signature,
+        );
+        let template_provider_config = pool_sv2::mining_pool::TemplateProviderConfig::new(
+            tp_address,
+            Some(authority_public_key),
+        );
+        let authority_config =
+            pool_sv2::mining_pool::AuthorityConfig::new(authority_public_key, authority_secret_key);
+        let config = Configuration::new(
+            connection_config,
+            template_provider_config,
+            authority_config,
+            coinbase_outputs,
+        );
+        let pool = PoolSv2::new(config);
+
+        Self { pool }
+    }
+}
+
+pub async fn start_template_provider(tp_port: u16) -> TemplateProvider {
+    let template_provider = TemplateProvider::start(tp_port);
+    template_provider.generate_blocks(16);
+    template_provider
+}
+
+pub async fn start_proxy(upstream: SocketAddr, downstream: SocketAddr) -> SV2Proxy {
+    let proxy = SV2Proxy::new();
+    let proxy_clone = proxy.clone();
+    tokio::spawn(async move {
+        proxy_clone.start(upstream, downstream).await;
+    });
+    proxy
+}
+
+pub async fn start_poolsv2(
```

Comment:
I guess this can be called simply `start_pool`?

Reply:
the original crate name is `pool_sv2`, I am just follow that